tests/impl.cpp

```diff
@@ -129,6 +129,36 @@ class SSE2NEONTestImpl : public SSE2NEONTest
                 }
             }
 
+#if SSE2NEON_PRECISE_MINMAX
+            if (test == it_mm_max_ps || test == it_mm_max_ss ||
+                test == it_mm_min_ps || test == it_mm_min_ss) {
+                // Make sure the NaN values are included in the testing
+                // one out of four times.
+                if ((rand() & 3) == 0) {
+                    uint32_t r1 = rand() & 3;
+                    uint32_t r2 = rand() & 3;
+                    mTestFloatPointer1[r1] = nanf("");
+                    mTestFloatPointer2[r2] = nanf("");
+                }
+            }
+
+            if (test == it_mm_max_pd || test == it_mm_max_sd ||
+                test == it_mm_min_pd || test == it_mm_min_sd) {
+                // Make sure the NaN values are included in the testing
+                // one out of four times.
+                if ((rand() & 3) == 0) {
+                    // FIXME:
+                    // The argument "0xFFFFFFFFFFFF" is a tricky workaround to
+                    // set the NaN value for doubles. The code is not intuitive
+                    // and should be fixed in the future.
+                    uint32_t r1 = ((rand() & 1) << 1) + 1;
```

Comment:
Ditto. I am not sure if it makes sense if `r1` and `r2` have the same value as indexing.